lib/util/mp4_parser.js

```diff
@@ -116,6 +116,10 @@ shaka.util.Mp4Parser = class {
     }
 
     let size = reader.readUint32();
+    if (!reader.hasMoreData()) {
+      this.done_ = true;
+      return;
+    }
```

Comment:
I don't get it. Earlier we check do we have 8 more bytes (size + type). On what circumstances we could have size only?
If this is a real life scenario, I think we should modify previous condition. And additionally, we should stop only when `stopOnPartial` is set, right?

Reply:
Updated according to our offline conversation